### DISCUSSION TITLE
Transferred Vigil of St. James: in Cistercian, it's B. Balduin, and I don't want to transfer the whole feast, only the Vigil... How?

### DISCUSSION THREAD

**Discussion by Augustinus-Altovadensis** [DivinumOfficium/divinum-officium#3960] [Q&A] [answered]
Laudétur Jesus Christus!

I'm gradually filling out all the Cistercian feasts and I've come up to an issue:
On July 24, in the Roman calendar, there is only the Vigil of St. James (with St. Christine commemorated), however in the Cistercian calendar, there is Bl. Balduin (iij. Lect. et M., it's like second class, definitely above the Vigil), with St. Christine commemorated. 

Now the question: if July 24 falls on Sunday, the Vigil, in my case the Commemoration of Vigil, should be transfered to Saturday, however the rest of the feasts, which become Commemorations, need to remain on Sunday. How to do that? 

Currently, if July 24 falls on Sunday, the entire` 07-24.txt` file gets transferred on Saturday, which is however not desirable. 

Should I use the `[Commemoratio Sabbat]` on July 23 and try to cancel the transfer of 07-24 to Saturday? Or is there a more elegant way to do that?

Thank you and God bless!

**Comment by FAJ-Munich** [comment, marked as answer]
Please consult the solution to the transfer of the Vigil of the Assumption when August 14 is a Sunday (Divino Afflatu). Same situation with S. Eusebius in that case. You should be able to mimic this with equal elegance.

**Reply by Augustinus-Altovadensis** [reply]
Thank you, that was it! Well, not S. Eusebius, as he is under the Vigil, but Tridentine Octave of St. Lawrence is the same story.

I have added a couple of tweaks, so that the vigil is not commemorated on Vespers or Sunday, so now it works, only on Saturday's Vespers on July 23, 2033, the Commemoration of St. Christine gets doubled for some reason. Any idea, where to look to fix that?

**Reply by FAJ-Munich** [reply]
Does the Solution to 08-22-2026 (transfer of Vigila S. Bartholomæi) help?
I am not sure about how you achieved the first part of your solution, so hard to tell where the double commemoration arises from.

**Reply by Augustinus-Altovadensis** [reply]
I did 3 things:
1. I split the Vigil and B. Balduin + Christine into two files.
2. I added following line into the Calendar file: `07-24=07-24cc~07-24=B. Balduini, Abbatis et Confessoris=2=Vigilia S. Jacobi Apostoli=1.5=S. Christinae, Virginis et Martyris=1=`. The "07-24cc" file contains B. Balduin and S. Christine, the 07-24 file just the Vigil. 
3. I added three lines in the `vigilia_commemoratio` function in` specials.pl`, that would take care for the unwanted Commemorations:
```
  if ($version =~ /Cistercien/ && $dayofweek == 0) { return ''; }
  if ($version =~ /Cistercien/ && $hora !~ /laudes/i) { return ''; }
  if ($version =~ /Cistercien/ && $winner{Rank} =~ /Vigilia/i) { return ''; } 
```
That's it. It does everything it's supposed to, just it doubles the Comm. of St. Christine for some reason. It obviously reads the [Commemoratio] tag in 07-24.txt twice, because B. Balduin's Commemoration is there just once, as it is supposed to. I tried to look for `$transferredvigil` in the code, but that didn't really bring anything helpful. I guess there needs to be one more check for `$transferredvigil`...

**Reply by FAJ-Munich** [reply]
On master, the transfer file for Dominical Letter *b*, by now has:
```
07-23=07-23~07-24;;1570 1888 1906 DA
```

Yours, given that you haven't merged the recent developments into your branch for a long time, still uses the slightly wonky tranfer of vigils. What could help is

```
07-23=07-23~07-24;;1570 1888 1906 DA
07-24=07-24cc;;1570 1888 1906 DA
```

But, imho, I think you definately need to consider merging your changes and the developments on `master`or there comes the point were it is completely incompatible. The sooner the better, I am sure, and I know from my one sub-branches: it is going to be painful but worth the effort if offered up to the Glory of the Lord!

+LJch+

**Reply by Augustinus-Altovadensis** [reply]
Thank you! It didn't really help, but at least I saw another part of the program, that I have never seen before. Currently, there is this line for year b: `07-23=07-24v;;1570 1888 1906 DA`.

This hasn't really come to mind, because there is no `07-24v` file in the `Sancti` folder...

Yes, rebase... About half a year ago, I did it, but of course, GitHub refused to even try to merge my changes into the upstream master, so I had to do it all again. It took me about a month to get everything working again, and some things still don't work... But of course, the Commemorations are quite a pickle, and the rules for the Cistercian Office are obviously different that for the Roman Office. A logical solution would be to include the Cistercian version in upstream, but I don't know, if you (the maintainers) are willing to go through that commit, because it would be quite a lot... However, currently about 95% of the technical stuff works, which is - as I understand - much more than in the Dominican version, yet that one is in the upstream. And currently I'm working on the Calendar, which is quite a lot of work, as a lot is different to both Tridentine and Monastic (1963) versions, and I'm at somewhere around 60%... 

To be honest, after rebasing, I kept my branch in sync with upstream for quite some time, but then someone reworked the handling of Antiphons, which is something I needed to change quite a lot, and I would have to devise a new solution to that function, so I just gave up. 

But as I said, I might try to do that, along with uploading the Cistercian version to the master - if it's realistic and practical for the maintainers. If you want to, you can look at the code, just searching for the word `Cistercien` will show pretty much all my changes.

**Reply by APMarcello3** [reply]
@Augustinus-Altovadensis I would have no problem with adding the Cistercian rite as it is now, if you are comfortable with that.

**Reply by FAJ-Munich** [reply]
@Augustinus-Altovadensis  

> If you want to, you can look at the code, just searching for the word Cistercien will show pretty much all my changes.

Happy to do so. Beforehand, however, I have two parts partially completed which I'd keep prioritised: Verse numberings #838 and at least the temporal for M1617 and M1930 and its associated code changes.
Given that I am going to be on Summer holidays from next week until the Assumption, I guess early September could be a good time, I can have a deeper look. In the meantime it would be great if you can share a link to some pdf scan of the Cistercian version you are implementing so I can start reading into the rubrics.

**Reply by Augustinus-Altovadensis** [reply]
Very good, thank you! Now the question: what should I do? Should I rebase on current master and then create a push, or will you do it manually? 

I will scan the rubrics and send here. The texts come from here: https://www.dropbox.com/scl/fo/fmnvuvxwpdxozz0pga0bx/AGi6rovWA0nOIhdWHuYPGuk?rlkey=68m6lmkt6p08wupo9zl4rlmg2&dl=0
However there are next to no rubrics...

**Reply by FAJ-Munich** [reply]
I would love to understand your code changes in the light of your sources: (once time permits) to see how the changes you have made can be blended in given the parallel development for other versions (M1617/1930) which may or may not have an overlap. That's why I also want to have a look at your sources: and the nice side effect is to actually gain more knowledge from that exercise.

**Reply by mbab** [reply]
> Very good, thank you! Now the question: what should I do? Should I rebase on current master and then create a push, or will you do it manually?

I suspect `rebase master` can be very hard as since Your last rebase there were: moves of data directories (Tabulae & Ordinarium), shuffling functions to new files and last but not least applying of `perl tidy`.
Simpler way can be adjust Your work step by step to new branch (pointed at current master) starting with:
- modifying horas.dialog
- add Kalendar/Transfer files
- add *Cist directories
- modifying Ordinarium (which is now one for all versions), with necessary additions to Prayers & Translate.

Then applying code changes, maybe hourwise.

_Of course it is only kind advice_

**Reply by Augustinus-Altovadensis** [reply]
@FAJ-Munich Very good. I have finally managed to scan the rubrics, you can find them here:
[Rubricæ Generales Breviarii Cisterciensis (1951).pdf](https://github.com/user-attachments/files/16539541/Rubricae.Generales.Breviarii.Cisterciensis.1951.pdf)

For the actual texts, I have already uploaded a Dropbox link with both parts of Antiphonary and a Psalter, which contains a complete Ferial Office, Hymns and Psalms of course. 

Now you should have everything you need to construct the 1951 Cistercian Divine Office. 

However, what I'm doing is not exactly 1951 Cistercian Office with the Common Calendar, but as the Vyšší Brod Abbey is the only monastery in the World that currently uses this, I also use their customs - their specific calendar and several rules cherrypicked from the 1957 General Chapter. The differences are minor, but there are some. Apart from the Calendar for Bohemia (and elevation of St. Anne Feast specific for Vyšší Brod only), the Sundays are elevated over MM. min., initial Pater, Ave and Credo (where applies) are abolished, the other Christmas Octaves are abolished, so are the Octaves of Saints (actually the 1957 Decree abolished all octaves apart from Christmas, Easter and Pentecost, but the Brothers have decided not to take this over completely). Also some Feasts have been taken over from the 1965 Supplement, but not the ranking of the old ones. That's about it. It's a bit complicated, but it makes sense for me to implement a version that actually someone uses (the Brothers and friends of Vyšší Brod Abbey), and not to do some theoretically clean version that no one in the world uses.

Hope that's all 🙂

**Reply by Augustinus-Altovadensis** [reply]
@mbab 
Yes, you are right, applying all the changes is a way to go. Last time, GitHub also told me that rebase is not possible, but I could manually merge the changes... So when I have some time, I will get on it. Last time it took me about two weeks to implement back all the changes, now it should be a bit faster, because (I hope) the `occurrence()` and `concurrence()` functions might not have changed that much, and the rest should be reasonably similar.

**Reply by FAJ-Munich** [reply]
Thanks a lot! Looking forward to read into it later this month. From what you describe, it sounds a bit like a "Reduced 1957" version similar to the Roman 1955. Let's see 😀

**Reply by Augustinus-Altovadensis** [reply]
My pleasure. Yes, the full 1957 Decree is like the Roman 1955, but we use only a fraction of all those "improvements". So it's still OK, nothing like the 1963 Monastic version 🙂
